CHANGES.rst

```diff
@@ -7,6 +7,20 @@ Dependencies
 Incompatible changes
 --------------------
 
+* LaTeX: The inclusion of ``amssymb`` LaTeX package is done after the contents
+  of :confval:`latex_elements`\ ``['fontpkg']``, and originates in the new
+  ``'fontpkgmath'`` key of :confval:`latex_elements`.
+
+  - Thus, move custom math font set-up from ``'fontpkg'`` to ``'fontpkgmath'``
+    if needed.
+  - Regarding Unicode latex engines, they do not use ``amssymb`` anymore but
+    XITS Math font via ``unicode-math``, as default of ``'fontpkgmath'``.
+    This fixes some issues and provides math glyphs much better matching GNU
+    FreeSerif which is the default text font.
+  - For ``'pdflatex'`` as :confval:`latex_engine`, it is planned to replace at
+    Sphinx 9 usage of ``amssymb`` by ``stix2`` in order for the math glyphs to
+    much better match the default Times-like text font.
```

Comment:
The reason for not doing it immediately is twofold:
- it will change looks of PDF,
- but is not motivated by any current bug to solve,
- and stix2 has an annoying note that current version of amsmath should be loaded after it but old versions should be loaded before it, which makes for a rather nightmarish preparation on our side to test what really happens say with people using TL2019/Debian LaTeX installation.

Besides, stix2 also changes the text font, there is an option for it not to, it is again a Times clone like tgtermes we use currently.

However stix2 is pretty much doing for `'pdflatex'` what unicode-math + XITS Math do for ``'xelatex'`` so there is motivation to use it.

Reply:
Maybe we should not be too affirmative on Sphinx 9 using `stix2` with `'pdflatex'`.  This package has a number of bugs, such as [stix2-throws-latex-font-warning-with-boldsymbol-or-bm](https://tex.stackexchange.com/questions/648668/stix2-throws-latex-font-warning-with-boldsymbol-or-bm), and using it will require probably a wrapper package to fix this and a number of other bugs [^1]. This is quite some work and we will have to check also situation for Japanese documents with `uplatex`.  I am not motivated to investigate this fully now and perhaps will not be in time for Sphinx 9.  And switching to `'xelatex'` with XITS Math as this PR makes default avoids having to deal with stix2 bugs.  It increases a bit the build time though.

[^1]: I suddenly remember I looked at the package in 2021 and reported https://github.com/stipub/stix_type1/issues/1, https://github.com/stipub/stix_type1/issues/2, https://github.com/stipub/stix_type1/issues/3, https://github.com/stipub/stix_type1/issues/4 (at least one of them is not for math mode) but the package looks (completely) unmaintained.

Reply:
No announcement is made for any Sphinx 9 change anymore for ``'pdflatex'``.